dbcon/joblist/groupconcat.cpp

```diff
@@ -331,16 +331,19 @@ void GroupConcatAgUM::initialize()
 
     fConcator->initialize(fGroupConcat);
 
-    fGroupConcat->fRowGroup.initRow(&fRow, true);
-    fData.reset(new uint8_t[fRow.getSize()]);
-    fRow.setData(fData.get());
+    bool useStringTable = fGroupConcat->fRowGroup.getRowSizeWithStrings() > 10 * (1 << 20);
+    fData.reinit(fGroupConcat->fRowGroup, 1, useStringTable);
+    fGroupConcat->fRowGroup.initRow(&fRow, useStringTable);
+    fData.getRow(0, &fRow);
 }
 
 
 void GroupConcatAgUM::processRow(const rowgroup::Row& inRow)
 {
     applyMapping(fGroupConcat->fMapping, inRow);
     fConcator->processRow(fRow);
+    if (fRow.usesStringTable() && fData.getStringTableMemUsage() > 50 * (1 << 20))
+        fData.clearStringStore();
```

Comment:
We should move this threshold value (50 * (1 << 20)) into a common location (maybe in rowgroup.h) as a const int or a macro, so it can be easily updated later.

Use of UNLIKELY around the if predicate might be a suitable optimization?

We are missing a line break before the if condition.